components/profiler/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Profile a part of the code using CPU Profiler from gperftools or Callgrind.
+//! Supports Linux and MacOS.
+//!
+//! ## Requirements
+//!
+//! 1. gperftools
+//!
+//!    Linux:
+//!
+//!      You can follow its [INSTALL manual](https://github.com/gperftools/gperftools/blob/master/INSTALL).
+//!      Roughly the instructions are the following:
+//!
+//!      1. Download packages from [release](https://github.com/gperftools/gperftools/releases)
+//!      2. Run `./configure`
+//!      3. Run `make install`
+//!
+//!    MacOS:
+//!
+//!      Simply `brew install gperftools`.
+//!
+//! ## Usage
+//!
+//! ```ignore
+//! profiler::start("./app.profile");
+//! some_complex_code();
+//! profiler::stop();
+//! ```
+//!
+//! Then, compile the code with `profiling` feature enabled.
+//!
+//! By default, a profile called `app.profile` will be generated by CPU Profiler.
+//! You can then analyze the profile using [pprof](https://github.com/google/pprof).
+//!
+//! If the application is running in Callgrind, a Callgrind profile dump will be generated instead.
+//! Notice that you should run Callgrind with command line option `--instr-atstart=no`, e.g.:
+//!
+//! ```bash
+//! valgrind --tool=callgrind --instr-atstart=no ./my_example
+//! ```
+//!
+//! Also see `examples/prime.rs`.
+
+#[allow(unused_extern_crates)]
+extern crate tikv_alloc;
```

Comment:
Why use `tikv_alloc`?

Reply:
tikv_alloc generally needs to be linked into every crate that doesn't link to tikv, so that tests and benches of that crate use tikv's allocator. I don't _think_ this extern crate statement is needed though as long as the dependency exists.

Reply:
CI will fail if alloc is not linked

Reply:
Ah, right. There are tests that all binaries contain jemalloc.